Update to latest

SubhraB opened the pull request.

ConfluentJenkins committed:
chore: update repo semaphore config

perdue committed:
Use schema ID returned from GetID() (fixes #848) (#870)

kkoehler committed:
Added SetRoundtripDuration function to Mockcluster to configure the round-trip duration (#892)

* added SetRoundtripDuration function to configure the round-trip duration for a single broker. this can be used for timeout testing

* added error handling to SetRoundtripDuration function

ConfluentJenkins committed:
chore: update repo semaphore project

ConfluentJenkins committed:
chore: update repo semaphore config

okulik committed:
Add info about arm64 statically built target (#900)

josvisser66 committed:
Go wrapper for rd_kafka_sasl_set_credentials (#879)

* Go wrapper for rd_kafka_sasl_set_credentials

* Fixes after code review

* Fixes after code review

* Better description in CHANGELOG for this change.

* Changes requested through code review

milindl committed:
Deprecate channel based clients (#894)

Adds Deprecated: to the godoc for ProduceChannel, and consumer's Event
method.

Adds BUG: to the godoc for Flush and Len to emphasize they don't play
well with ProduceChannel.

Change examples and changelog.

snosratiershad committed:
docs: correct consumer example code in README (#846)


Signed-off-by: Salar Nosrati-Ershad <snosratiershad@gmail.com>
Co-authored-by: Milind L <miluthra@confluent.io>

milindl committed:
Add IsTimeout method to Error (#903)

This convenience method just checks if Error code is
ErrTimedOut/ErrTimedOutQueue.

Also modifies the example in README.md.

Modifies the CHANGELOG.md.

milindl committed:
Fix gofmt errors and broker check on integration test (#904)

An integration test which should be checking for testconf.json, was not
checking for it correctly, and wasn't being skipped due to that.

There were several issues with formatting, all of them in the generated
error file. All those have been fixed along with the generation code.

Additionally, there are a few errors in the _build files, but they won't
be fixed till we bump our go version to 1.17 (+build is replaced with
go:build, and gofmt 1.18+ expects go:build, but we are going to stick to
go 1.14 for now).

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/Kong/confluent-kafka-go?pullRequest=2) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/Kong/confluent-kafka-go?pullRequest=2) before we can accept your contribution.<br/>**0** out of **7** committers have signed the CLA.<br/><br/>:x: perdue<br/>:x: kkoehler<br/>:x: okulik<br/>:x: ConfluentJenkins<br/>:x: josvisser66<br/>:x: snosratiershad<br/>:x: milindl<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/Kong/confluent-kafka-go?pullRequest=2) it.</sub>

milindl committed:
[producer] Fix TestTransactionalAPI for InitTransactions changes (#908)

A recent refactor of the underlying init_transactions code in terms of
timeouts and the ability to retry requires changes in
TestTransactionalAPI test.

milindl committed:
[consumer] Ignore Seek() timeout (#906)

Seek timeout is ignored and an infinite timeout is used instead.
The method will block till fetcher state is updated and the seek is
registered.

See #888.

milindl committed:
Add Github Actions setup (#905)

1. build, test
2. gofmt
3. make-docs
